File: apple/MeshDropMac/Sources/mock/MockData.swift
import Foundation
import SwiftUI

// MARK: ─── 设备 ─────────────────────────────────────────
//
// 这里的 `Mock*` 类型 (`MockDevice`, `MockHistory`, etc.) 是 view DTO，runtime 由
// `ShareEngine` 模型经 adapter 投影成这些 record，给 SwiftUI 复用。
// `static let all` 等静态数据数组仅供 Xcode Preview / 离线截图用 —— 已用 `#if DEBUG`
// 围起来，release build 不含；运行时的数据全部来自 engine。

enum DeviceKind: String, CaseIterable {
    case mac, win, ios, android, ipad
}

struct MockDevice: Identifiable, Hashable {
    let id: String
    let name: String          // "Lily's MacBook"
    let who: String           // "李莉"
    let kind: DeviceKind
    let dist: Double          // 雷达半径 0-1
    let angle: Double         // 角度 0-360
    let color: Color          // initials 圆形彩色背景
    let initials: String      // "LL"
    let os: String            // "macOS"
    let rtt: Int              // ms
    let online: Bool

#if DEBUG
    static let all: [MockDevice] = [
        .init(id: "lily",    name: "Lily's MacBook",   who: "李莉",   kind: .mac,     dist: 0.55, angle: 35,  color: Color(hex: 0xFFB4A1), initials: "LL", os: "macOS",  rtt: 18, online: true),
        .init(id: "kun",     name: "Kun · Pixel 8",    who: "坤",     kind: .android, dist: 0.78, angle: 110, color: Color(hex: 0xB7E5C8), initials: "K",  os: "Pixel",  rtt: 32, online: true),
        .init(id: "jiawei",  name: "Jiawei · iPad",    who: "嘉伟",   kind: .ipad,    dist: 0.40, angle: 200, color: Color(hex: 0xC7B8FF), initials: "JW", os: "iPadOS", rtt: 14, online: true),
        .init(id: "mengxi",  name: "Meng Xi · iPhone", who: "孟茜",   kind: .ios,     dist: 0.62, angle: 265, color: Color(hex: 0xFFD970), initials: "MX", os: "iOS",    rtt: 26, online: true),
        .init(id: "dev01",   name: "DEV-01 · Win 11",  who: "工位机", kind: .win,     dist: 0.88, angle: 320, color: Color(hex: 0x9AD0FF), initials: "D1", os: "Win 11", rtt: 41, online: true),
    ]
#endif
}

// MARK: ─── 历史 ─────────────────────────────────────────

enum HistoryDir { case incoming, outgoing }
enum HistoryKind: String { case image, file, text }
enum HistoryStatus: String { case done, transferring, queued, failed }

struct MockHistory: Identifiable {
    let id: String
    let dir: HistoryDir
    let peer: String
    let time: String
    let kind: HistoryKind
    var name: String? = nil
    var size: String? = nil
    var ext: String? = nil
    var fileURL: URL? = nil
    var content: String? = nil
    var count: Int? = nil
    var progress: Int? = nil
    let status: HistoryStatus

#if DEBUG
    static let all: [MockHistory] = [
        .init(id: "h6", dir: .incoming, peer: "孟茜", time: "14:18", kind: .image,                                                        count: 2,            status: .done),
        .init(id: "h5", dir: .outgoing, peer: "孟茜", time: "14:10", kind: .file,  name: "设计稿_v3_final.fig",     size: "14.2 MB", ext: "fig",                       status: .done),
        .init(id: "h4", dir: .outgoing, peer: "李莉", time: "14:09", kind: .text,                                                                 content: "改完了，整理一下发你 👇", status: .done),
        .init(id: "h3", dir: .outgoing, peer: "嘉伟", time: "14:08", kind: .file,  name: "iOS-mocks-final.zip",    size: "48.6 MB", ext: "zip", progress: 67,           status: .transferring),
        .init(id: "h2", dir: .incoming, peer: "坤",   time: "13:58", kind: .file,  name: "IMG_4821~38.heic",       size: "128 MB",  ext: "heic", progress: 12,          status: .transferring),
        .init(id: "h1", dir: .outgoing, peer: "李莉", time: "13:42", kind: .file,  name: "demo-video.mp4",         size: "512 MB",  ext: "mp4",                         status: .queued),
    ]
#endif
}

// MARK: ─── 待审 ─────────────────────────────────────────

struct MockPendingPairing {
    let id: String
    let peer: String
    let deviceName: String
    let fingerprint: String
    let receivedAt: String

    static let sample = MockPendingPairing(
        id: "pp-1",
        peer: "李莉",
        deviceName: "Lily's MacBook",
        fingerprint: "ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF",
        receivedAt: "8s ago"
    )
}

struct MockPendingOffer {
    let id: String
    let peer: String
    let deviceName: String
    let fileName: String
    let fileSize: String
    var isImage: Bool = false
    var previewBase64: String? = nil
    let note: String
    let receivedAt: String

    static let sample = MockPendingOffer(
        id: "po-1",
        peer: "嘉伟",
        deviceName: "Jiawei · iPad",
        fileName: "规划文档_v0.3.pages",
        fileSize: "3.4 MB",
        note: "改完了帮我看下第二章，特别是 §2.3 那段",
        receivedAt: "刚刚"
    )
}

// MARK: ─── 剪贴板 ───────────────────────────────────────

enum ClipKind: String { case link, text, code }

struct MockClip: Identifiable {
    let id: String
    let who: String
    let kind: ClipKind
    let body: String
    let lang: String?
    let ago: String

#if DEBUG
    static let all: [MockClip] = [
        .init(id: "cb1", who: "嘉伟", kind: .link, body: "https://internal.acme.io/specs/auth-v3", lang: nil, ago: "8s"),
        .init(id: "cb2", who: "孟茜", kind: .text, body: "1. 新流程要支持端到端\n2. 雷达扫描频率调到 2s\n3. iPad 端做横屏适配", lang: nil, ago: "12m"),
        .init(id: "cb3", who: "李莉", kind: .code, body: "docker run --rm -v $PWD:/app meshdrop/build:latest", lang: "sh", ago: "34m"),
        .init(id: "cb4", who: "坤",   kind: .text, body: "会议室 B 已订到 16:00–17:30", lang: nil, ago: "1h"),
        .init(id: "cb5", who: "我",   kind: .link, body: "figma://file/Q8xK2/MeshDrop?node-id=42:108", lang: nil, ago: "2h"),
    ]
#endif
}

// MARK: ─── 传输 ─────────────────────────────────────────

enum TransferState: String { case sending, receiving, done, queued, failed }

struct MockTransfer: Identifiable {
    // 真实数据投影时传入 history.id；MockData preview 自动给新 UUID
    var id: UUID = UUID()
    let name: String
    let size: String
    let ext: String
    let from: String
    let to: String
    let progress: Int
    let state: TransferState
    let speed: String?
    let eta: String?
    /// 失败原因（校验失败 / 连接中断 / 对方拒收 …），仅 state == .failed 时有值。
    var failReason: String? = nil

#if DEBUG
    static let all: [MockTransfer] = [
        .init(name: "设计稿_v3_final.fig",     size: "14.2 MB",          ext: "fig",  from: "我", to: "孟茜",  progress: 100, state: .done,      speed: nil,            eta: "00:08"),
        .init(name: "iOS-mocks-final.zip",     size: "48.6 MB",          ext: "zip",  from: "我", to: "孟茜",  progress: 67,  state: .sending,   speed: "8.4 MB/s",      eta: "00:02"),
        .init(name: "spec_PRD_2026Q1.pdf",     size: "2.1 MB",           ext: "pdf",  from: "我", to: "嘉伟",  progress: 34,  state: .sending,   speed: "3.1 MB/s",      eta: "00:01"),
        .init(name: "IMG_4821~IMG_4838.heic",  size: "128 MB · 18 张",   ext: "heic", from: "坤", to: "我",    progress: 12,  state: .receiving, speed: "11.7 MB/s",     eta: "00:09"),
        .init(name: "release-notes.md",        size: "4.8 KB",           ext: "md",   from: "我", to: "DEV-01", progress: 100, state: .done,     speed: nil,            eta: "00:01"),
        .init(name: "demo-video.mp4",          size: "512 MB",           ext: "mp4",  from: "我", to: "李莉",  progress: 0,   state: .queued,    speed: nil,            eta: nil),
    ]
#endif
}

// MARK: ─── 速度 ─────────────────────────────────────────

enum MockSpeed {
    static let uploadBars:   [Int] = [3, 5, 8, 7, 9, 6, 11, 12, 14, 11, 10, 11, 12, 11]
    static let downloadBars: [Int] = [8, 9, 7, 6, 5, 7, 10, 12, 11, 12, 11, 12, 11, 12]
    static let sessionBars:  [Int] = [2, 3, 5, 4, 6, 8, 7, 9, 10, 12, 11, 12, 11, 12, 14]
}

// MARK: ─── 本机 ─────────────────────────────────────────

struct MockMe {
    static let name = "我"
    static let deviceName = "This MacBook"
    static let fingerprint = "ZX8K · L72M · 9FQ3 · 7HD2"
    static let fullFingerprint = "ZX8K · L72M · 9FQ3 · 7HD2 · M1P6 · QA8N · KZ9R · X3WF"
    static let ip = "192.168.1.42"
    static let os = "macOS"
    static let visibility = "可见 · Visible · LAN"
}

// MARK: ─── 配对设备表 ───────────────────────────────────

struct MockTrustedDevice: Identifiable {
    let id: String
    let name: String
    let who: String
    let kind: DeviceKind
    let fingerprint: String
    let pairedAt: String
    let lastSeen: String
    let online: Bool

#if DEBUG
    static let all: [MockTrustedDevice] = [
        .init(id: "t1", name: "Lily's MacBook",   who: "李莉", kind: .mac,     fingerprint: "ZX8K · L72M · 9FQ3 · 7HD2", pairedAt: "2026-04-12", lastSeen: "刚刚",       online: true),
        .init(id: "t2", name: "Meng Xi · iPhone", who: "孟茜", kind: .ios,     fingerprint: "P4R7 · KQ2X · L9MN · 7CA1", pairedAt: "2026-03-28", lastSeen: "刚刚",       online: true),
        .init(id: "t3", name: "Jiawei · iPad",    who: "嘉伟", kind: .ipad,    fingerprint: "9HX2 · BNT4 · LM7Q · K3FZ", pairedAt: "2026-03-11", lastSeen: "刚刚",       online: true),
        .init(id: "t4", name: "Kun · Pixel 8",    who: "坤",   kind: .android, fingerprint: "C8YR · NMP3 · X1QK · L8H9", pairedAt: "2026-02-17", lastSeen: "2 天前",    online: false),
        .init(id: "t5", name: "DEV-01 · Win 11",  who: "工位机", kind: .win,   fingerprint: "T2W8 · KR5N · MX9P · BLH6", pairedAt: "2026-01-09", lastSeen: "刚刚",       online: true),
        .init(id: "t6", name: "MBP-2019",         who: "我",   kind: .mac,     fingerprint: "QZ7X · 8RTM · K3PN · 9LWY", pairedAt: "2025-11-22", lastSeen: "5 天前",    online: false),
    ]
#endif
}

File: android/app/src/main/java/com/welape/meshdrop/mock/MockData.kt
package com.welape.meshdrop.mock

import androidx.compose.ui.graphics.Color
import com.welape.meshdrop.ui.theme.AvatarLilac
import com.welape.meshdrop.ui.theme.AvatarMint
import com.welape.meshdrop.ui.theme.AvatarPeach
import com.welape.meshdrop.ui.theme.AvatarSky
import com.welape.meshdrop.ui.theme.AvatarSun

/**
 * UI-FIRST 阶段所有静态展示数据。
 * 与 backend 完全解耦；下一轮接 [com.welape.meshdrop.transport.ShareEngine] 时再替换。
 */

enum class DeviceKind { MAC, IPHONE, IPAD, ANDROID, WIN, LINUX }

data class MockDevice(
    val id: String,
    val name: String,
    val who: String,
    val kind: DeviceKind,
    val dist: Float,         // 0..1，雷达极坐标
    val angleDeg: Int,       // 0..360
    val color: Color,
    val initials: String,
    val os: String,
    val rttMs: Int,
    val online: Boolean = true,
    val ip: String,
    /** 会话键：对端指纹（跨会话/离线稳定）。mock 样本留空。 */
    val fingerprint: String = "",
)

val MockDevices: List<MockDevice> = listOf(
    MockDevice(
        id = "lily", name = "Lily's MacBook", who = "李莉", kind = DeviceKind.MAC,
        dist = 0.55f, angleDeg = 35, color = AvatarPeach, initials = "LL",
        os = "macOS", rttMs = 18, ip = "192.168.1.18",
    ),
    MockDevice(
        id = "kun", name = "Kun · Pixel 8", who = "坤", kind = DeviceKind.ANDROID,
        dist = 0.78f, angleDeg = 110, color = AvatarMint, initials = "K",
        os = "Pixel", rttMs = 32, ip = "192.168.1.27",
    ),
    MockDevice(
        id = "jiawei", name = "Jiawei · iPad", who = "嘉伟", kind = DeviceKind.IPAD,
        dist = 0.40f, angleDeg = 200, color = AvatarLilac, initials = "JW",
        os = "iPadOS", rttMs = 14, ip = "192.168.1.31",
    ),
    MockDevice(
        id = "mengxi", name = "Meng Xi · iPhone", who = "孟茜", kind = DeviceKind.IPHONE,
        dist = 0.62f, angleDeg = 265, color = AvatarSun, initials = "MX",
        os = "iOS", rttMs = 26, ip = "192.168.1.45",
    ),
    MockDevice(
        id = "dev01", name = "DEV-01 · Win 11", who = "工位机", kind = DeviceKind.WIN,
        dist = 0.88f, angleDeg = 320, color = AvatarSky, initials = "D1",
        os = "Win 11", rttMs = 41, ip = "192.168.1.55",
    ),
)

// MARK: 历史

enum class HistoryDir { INCOMING, OUTGOING }
enum class HistoryStatus { DONE, TRANSFERRING, QUEUED, FAILED }

sealed interface HistoryKindMock {
    data class Image(val count: Int) : HistoryKindMock
    data class File(val name: String, val size: String, val ext: String, val progress: Int? = null) : HistoryKindMock
    data class Text(val content: String) : HistoryKindMock
}

data class MockHistoryItem(
    val id: String,
    val dir: HistoryDir,
    val peer: String,
    val time: String,
    val kind: HistoryKindMock,
    val status: HistoryStatus,
    /** 该条目对端的会话键（指纹），供历史页点开跳转会话用。 */
    val peerKey: String = "",
)

val MockHistory: List<MockHistoryItem> = listOf(
    MockHistoryItem("h6", HistoryDir.INCOMING, "孟茜", "14:18", HistoryKindMock.Image(2), HistoryStatus.DONE),
    MockHistoryItem("h5", HistoryDir.OUTGOING, "孟茜", "14:10", HistoryKindMock.File("设计稿_v3_final.fig", "14.2 MB", "fig"), HistoryStatus.DONE),
    MockHistoryItem("h4", HistoryDir.OUTGOING, "李莉", "14:09", HistoryKindMock.Text("改完了，整理一下发你 ↓"), HistoryStatus.DONE),
    MockHistoryItem("h3", HistoryDir.OUTGOING, "嘉伟", "14:08", HistoryKindMock.File("iOS-mocks-final.zip", "48.6 MB", "zip", progress = 67), HistoryStatus.TRANSFERRING),
    MockHistoryItem("h2", HistoryDir.INCOMING, "坤", "13:58", HistoryKindMock.File("IMG_4821~38.heic", "128 MB", "heic", progress = 12), HistoryStatus.TRANSFERRING),
    MockHistoryItem("h1", HistoryDir.OUTGOING, "李莉", "13:42", HistoryKindMock.File("demo-video.mp4", "512 MB", "mp4"), HistoryStatus.QUEUED),
)

// MARK: 配对 / 文件 offer

data class MockPendingPairing(
    val id: String = "pp-1",
    val peer: String = "李莉",
    val deviceName: String = "Lily's MacBook",
    val fingerprintGroups: List<String> = listOf("ZX8K", "L72M", "9FQ3", "7HD2", "M1P6", "QA8N", "KZ9R", "X3WF"),
    val pinCode: String = "8-2-4-1-3-7",
    val receivedAt: String = "8s ago",
)

val MockPendingPairingItem = MockPendingPairing()

data class MockPendingOffer(
    val id: String = "po-1",
    val peer: String = "嘉伟",
    val deviceName: String = "Jiawei · iPad",
    val fileName: String = "规划文档_v0.3.pages",
    val fileSize: String = "3.4 MB",
    val note: String = "改完了帮我看下第二章，特别是 §2.3 那段",
    val receivedAt: String = "just now",
)

val MockPendingOfferItem = MockPendingOffer()

// MARK: 剪贴板

enum class ClipKind { LINK, TEXT, CODE }

data class MockClip(
    val id: String,
    val who: String,
    val kind: ClipKind,
    val body: String,
    val ago: String,
    val lang: String? = null,
)

val MockClips: List<MockClip> = listOf(
    MockClip("cb1", "嘉伟", ClipKind.LINK, "https://internal.acme.io/specs/auth-v3", "8s"),
    MockClip("cb2", "孟茜", ClipKind.TEXT, "1. 新流程要支持端到端\n2. 雷达扫描频率调到 2s\n3. iPad 端做横屏适配", "12m"),
    MockClip("cb3", "李莉", ClipKind.CODE, "docker run --rm -v \$PWD:/app meshdrop/build:latest", "34m", lang = "sh"),
    MockClip("cb4", "坤", ClipKind.TEXT, "会议室 B 已订到 16:00-17:30", "1h"),
    MockClip("cb5", "我", ClipKind.LINK, "figma://file/Q8xK2/MeshDrop?node-id=42:108", "2h"),
)

// MARK: 传输

enum class TransferState { DONE, SENDING, RECEIVING, QUEUED, FAILED }

data class MockTransfer(
    val name: String,
    val size: String,
    val ext: String,
    val from: String,
    val to: String,
    val progress: Int,
    val state: TransferState,
    val speed: String? = null,
    val eta: String? = null,
    val id: String = "",
    /** 已接收完成项的本地落盘 file:// Uri — 用于 OPEN 按钮调 ACTION_VIEW。 */
    val savedFileUri: String? = null,
    /** 失败原因（校验失败 / 连接中断 / 对方拒收 …），仅 state == FAILED 时有值。 */
    val failReason: String? = null,
)

val MockTransfers: List<MockTransfer> = listOf(
    MockTransfer("设计稿_v3_final.fig", "14.2 MB", "fig", "我", "孟茜", 100, TransferState.DONE, eta = "00:08"),
    MockTransfer("iOS-mocks-final.zip", "48.6 MB", "zip", "我", "孟茜", 67, TransferState.SENDING, "8.4 MB/s", "00:02"),
    MockTransfer("spec_PRD_2026Q1.pdf", "2.1 MB", "pdf", "我", "嘉伟", 34, TransferState.SENDING, "3.1 MB/s", "00:01"),
    MockTransfer("IMG_4821~IMG_4838.heic", "128 MB · 18 张", "heic", "坤", "我", 12, TransferState.RECEIVING, "11.7 MB/s", "00:09"),
    MockTransfer("release-notes.md", "4.8 KB", "md", "我", "DEV-01", 100, TransferState.DONE, eta = "00:01"),
    MockTransfer("demo-video.mp4", "512 MB", "mp4", "我", "李莉", 0, TransferState.QUEUED),
)

// 速度柱状图（采样）
val MockUploadBars = listOf(3, 5, 8, 7, 9, 6, 11, 12, 14, 11, 10, 11, 12, 11)
val MockDownloadBars = listOf(8, 9, 7, 6, 5, 7, 10, 12, 11, 12, 11, 12, 11, 12)
val MockSessionBars = listOf(2, 3, 5, 4, 6, 8, 7, 9, 10, 12, 11, 12, 11, 12, 14)

// MARK: 本机信息（自卡）

data class MockMe(
    val name: String = "Pixel 8 · 我",
    val fingerprintGroups: List<String> = listOf("ZX8K", "L72M", "9FQ3", "7HD2"),
    val ip: String = "192.168.1.42",
    val os: String = "Android 14",
    val visibility: String = "可见 · VISIBLE",
)

val MockMeData = MockMe()

// MARK: 聊天消息（与孟茜对话流）

enum class MsgSide { IN, OUT }
enum class MsgKind { TEXT, FILE, IMAGE }
enum class MsgState { SENT, DELIVERED, FAILED }

data class MockMessage(
    val id: String,
    val side: MsgSide,
    val kind: MsgKind,
    val time: String,
    val text: String? = null,
    val fileName: String? = null,
    val fileSize: String? = null,
    val fileExt: String? = null,
    val state: MsgState = MsgState.DELIVERED,
    val progress: Int? = null,
)

val MockChatWithMengxi = listOf(
    MockMessage("m1", MsgSide.IN, MsgKind.TEXT, "14:02",
        text = "刚做完 PRD 草稿，发你看看 →"),
    MockMessage("m2", MsgSide.IN, MsgKind.FILE, "14:03",
        fileName = "spec_PRD_2026Q1.pdf", fileSize = "2.1 MB", fileExt = "pdf"),
    MockMessage("m3", MsgSide.OUT, MsgKind.TEXT, "14:05",
        text = "收到，§2.3 那段要不要再细化一下？"),
    MockMessage("m4", MsgSide.OUT, MsgKind.TEXT, "14:06",
        text = "我把改完的发你 👇"),
    MockMessage("m5", MsgSide.OUT, MsgKind.FILE, "14:10",
        fileName = "设计稿_v3_final.fig", fileSize = "14.2 MB", fileExt = "fig",
        state = MsgState.DELIVERED),
    MockMessage("m6", MsgSide.OUT, MsgKind.FILE, "14:11",
        fileName = "iOS-mocks-final.zip", fileSize = "48.6 MB", fileExt = "zip",
        state = MsgState.SENT, progress = 67),
    MockMessage("m7", MsgSide.IN, MsgKind.TEXT, "14:18",
        text = "完美，正在下载 ✓"),
)

// 聊天列表（与各设备的会话）
data class MockChatPreview(
    val deviceId: String,
    val lastSnippet: String,
    val lastTime: String,
    val unread: Int = 0,
    val isFile: Boolean = false,
    /** 对端显示名，设备离线（不在 devicesUi）时用它还原会话行抬头。 */
    val peerName: String = "",
)

fun MockDeviceById(id: String): MockDevice? = MockDevices.firstOrNull { it.id == id }

val MockChatPreviews = listOf(
    MockChatPreview("mengxi", "完美，正在下载 ✓", "14:18", unread = 2),
    MockChatPreview("jiawei", "iOS-mocks-final.zip · 67%", "14:08", isFile = true),
    MockChatPreview("kun", "IMG_4821~38.heic · 接收中", "13:58", unread = 1, isFile = true),
    MockChatPreview("lily", "改完了，整理一下发你 ↓", "13:42"),
    MockChatPreview("dev01", "release-notes.md · 已送达", "12:30", isFile = true),
)

// MARK: 信任列表（已配对设备）

data class MockTrustRecord(
    val deviceName: String,
    val owner: String,
    val os: String,
    val fingerprintGroups: List<String>,
    val pairedDays: Int,
    val lastSeen: String,
)

val MockTrustList = listOf(
    MockTrustRecord(
        deviceName = "Lily's MacBook", owner = "李莉", os = "macOS Sonoma",
        fingerprintGroups = listOf("ZX8K", "L72M", "9FQ3", "7HD2"),
        pairedDays = 42, lastSeen = "8m ago",
    ),
    MockTrustRecord(
        deviceName = "Jiawei · iPad", owner = "嘉伟", os = "iPadOS 18",
        fingerprintGroups = listOf("Q8WK", "M6N3", "2FXP", "8AD4"),
        pairedDays = 12, lastSeen = "just now",
    ),
    MockTrustRecord(
        deviceName = "Meng Xi · iPhone", owner = "孟茜", os = "iOS 18",
        fingerprintGroups = listOf("V3RN", "8KK4", "ZW7Q", "9P22"),
        pairedDays = 5, lastSeen = "刚刚",
    ),
    MockTrustRecord(
        deviceName = "DEV-01 · Win 11", owner = "工位机", os = "Win 11 23H2",
        fingerprintGroups = listOf("K3RM", "6QPN", "M8X2", "7YH5"),
        pairedDays = 88, lastSeen = "12m ago",
    ),
)
